File: Assignment3/Client/src/opCode.cpp
//
// Created by Tomer Ofek on 28/12/2021.
//

#include "../include/opCode.h"

opCode::opCode() {}
short opCode::getOpCode(string action) {
    if(action=="REGISTER")
        return 1;
    if(action=="LOGIN")
        return 2;
    if(action=="LOGOUT")
        return 3;
    if(action=="FOLLOW")
        return 4;
    if(action=="POST")
        return 5;
    if(action=="PM")
        return 6;
    if(action=="LOGSTAT")
        return 7;
    if(action=="STAT")
        return 8;
    if(action=="NOTIFICATION")
        return 9;
    if(action=="ACK")
        return 10;
    if(action=="ERROR")
        return 11;
    if(action=="BLOCK")
        return 12;
    return -1;
}

string opCode::getAction(short opcode) {
    if (opcode == 9) return "NOTIFICATION";
    else if (opcode == 10) return "ACK";
    else if (opcode == 11) return "ERROR";
    else return "Nothing";
}

File: Assignment3/Server/src/main/java/bgu/spl/net/serverLogic/opCode.java
package bgu.spl.net.serverLogic;

public class opCode {
    public static final String REGISTER="01";
    public static final String LOGIN ="02";
    public static final String LOGOUT ="03";
    public static final String FOLLOW="04";
    public static final String POST="05";
    public static final String PM="06";
    public static final String LOGSTAT="07";
    public static final String STAT="08";
    public static final String NOTIFICATION="09";
    public static final String ACK="10";
    public static final String ERROR="11";
    public static final String BLOCK="12";

    public static String ErrorMsg(String x)
    {
        return "ERROR"+x;
    }
    public static String AckMsg(String x)
    {
        return "ACK"+x;
    }
    public static String SpecialAckMsg(String x,String content)
    {
        return "ACK"+x+content;
    }
    public static String notificationMsg(String p,String postingUser,String content)
    {
        return "NOTIFICATION"+p+postingUser+" "+content;
    }

    public static String notificationMsg(String p,String postingUser,String content,String date)
    {
        return "NOTIFICATION"+p+postingUser+" "+content+ " "+ date;
    }
}
